libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -1354,14 +1354,21 @@ void test_MQTT_DeserializeAck_pingresp( void )
     uint8_t buffer[ 10 ] = { 0 };
 
     /* Bad remaining length. */
+    ( void ) memset( ( void * ) &mqttPacketInfo, 0x00, sizeof( mqttPacketInfo ) );
     mqttPacketInfo.type = MQTT_PACKET_TYPE_PINGRESP;
-    mqttPacketInfo.pRemainingData = buffer;
     mqttPacketInfo.remainingLength = MQTT_PACKET_PINGRESP_REMAINING_LENGTH + 1;
     status = MQTT_DeserializeAck( &mqttPacketInfo, &packetIdentifier, &sessionPresent );
     TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
 
-    /* Process a valid PINGRESP. */
+    /* Process a valid PINGRESP with remaining data buffer NULL. */
+    mqttPacketInfo.remainingLength = MQTT_PACKET_PINGRESP_REMAINING_LENGTH;
+    mqttPacketInfo.pRemainingData = NULL;
+    status = MQTT_DeserializeAck( &mqttPacketInfo, NULL, NULL );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+
+    /* Process a valid PINGRESP with remaining data buffer non-NULL. */
     mqttPacketInfo.remainingLength = MQTT_PACKET_PINGRESP_REMAINING_LENGTH;
+    mqttPacketInfo.pRemainingData = buffer;
```

Comment:
Should such a case of 0 remaining length value but a valid buffer be supported as a valid input parameter?

Reply:
We are not enforcing the buffer to be NULL for PINGRESP.  A non-NULL buffer pointer will just be ignored in this case

Reply:
As discussed, I agree that this additional test is not required and will be giving a wrong indication. Updated the PR to remove the test.

Reply:
We discussed offline and decided that this test case is not required as it does not represent the intended use of the API.